src/storage/filesystem-adapter.test.ts

```
@@ -0,0 +1,150 @@
+      const testData = JSON.stringify({ test: 'data' });
+      await adapter.save(testData);
+
+      const fileContent = await fs.readFile(testPath, 'utf8');
+      expect(fileContent).toBe(testData);
+    });
+
+    it('should handle save errors gracefully', async () => {
+      // Create adapter with invalid path (using a file as directory)
+      const invalidPath = path.join(testPath, 'invalid.json');
+      await fs.writeFile(testPath, ''); // Create a file instead of directory
+      
```

ESLint

[error] Trailing spaces not allowed.

(no-trailing-spaces)